src/ctelnet.cpp

```diff
@@ -1603,27 +1597,23 @@ int cTelnet::decompressBuffer( char *& in_buffer, int& length, char* out_buffer
         inflateEnd( & mZstream );
         qDebug() << "recv Z_STREAM_END, ending compression";
         this->mNeedDecompression = false;
-        this->mMCCP_version_1 = false;
-        this->mMCCP_version_2 = false;
-
-        // Reset the option state so we can re-enable compression again in the future
-        // such as in the case of a copyover -JM
-        hisOptionState[static_cast<int>(OPT_COMPRESS)] = false;
-        hisOptionState[static_cast<int>(OPT_COMPRESS2)] = false;
```

Comment:
As I understand it the default state for any Telnet sub-option must be false so your:

>Thing is though when those option states are false they never go back to true, and is unneccesary to keep an eye on them...

is puzzling to me - but is that is _my_ problem? 😖 Or should we retain just those two lines (and the comment)?